### DISCUSSION TITLE
Webpack 5 build process hangs without any logs or info

### DISCUSSION THREAD

**Discussion by shirshendubhowmick** [webpack/webpack#14464] [Q&A] [answered]
I am using webpack 5. I facing an issue where webpack build process hangs at ` building 5/6 entries 12443/12532.......` without giving much log or info on the console.

I have identified this issue happened after I have updated an internal npm package which part of the application I am building. 
The package is basically a CSS library and the changes that were done was some refactors.

However I am still not sure what exactly inside the package is causing webpack to hang.

Can anyone suggest how to debug this issue ? I mean is there anyway I can get more logs on the build process with which I can identify whats going wrong ?

**Comment by alexander-akait** [comment]
How big your added file?

**Reply by shirshendubhowmick** [reply]
The package is already part of the application. It just that I made some changes and published a new version. (Will add this info in the original question)

The changes are basically some refactors without.

And this package is basically a CSS lib. I am guessing that the issue is probably caused because something going wrong with the loaders.

* style loader
* css loader
* postcss loader

These are the loaders that are being used currently.

**Reply by alexander-akait** [reply]
Sorry, need example to debug it, hard to say

**Reply by shirshendubhowmick** [reply]
Ok can you tell what kind of example you need ?

Like the webpack config or the changes that was introduced to that package ?

**Reply by alexander-akait** [reply]
Do you have github repo?

**Reply by shirshendubhowmick** [reply]
Unfortunately thats a private repo, which i won't be able to share publicly.

However I can share some code snippets.

Also if there is anyway I can generate more logs on webpack build that will also help I guess.

**Reply by alexander-akait** [reply]
as minimum run `npx webpack-cli info` and provide configuration(s) (or allowed part of configuration(s))

**Reply by shirshendubhowmick** [reply]
`npx webpack-cli info` details:

```
  System:
    OS: macOS 11.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 32.82 MB / 16.00 GB
  Binaries:
    Node: 14.17.0 - ~/.nvm/versions/node/v14.17.0/bin/node
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.0/bin/npm
  Browsers:
    Chrome: 94.0.4606.81
    Edge: 94.0.992.47
    Firefox: 92.0
    Firefox Developer Edition: 80.0
    Safari: 15.0
  Packages:
    babel-loader: ^8.0.6 => 8.2.2 
    clean-webpack-plugin: ^4.0.0 => 4.0.0 
    copy-webpack-plugin: ^9.0.1 => 9.0.1 
    css-loader: ^3.4.2 => 3.6.0 
    html-loader: ^0.5.5 => 0.5.5 
    html-webpack-plugin: ^5.3.2 => 5.3.2 
    postcss-loader: ^6.1.0 => 6.1.1 
    string-replace-loader: ^3.0.2 => 3.0.3 
    style-loader: ^1.1.3 => 1.3.0 
    terser-webpack-plugin: ^5.2.4 => 5.2.4 
    webpack: ^5.40.0 => 5.57.1 
    webpack-bundle-analyzer: ^4.4.2 => 4.4.2 
    webpack-cli: ^4.7.2 => 4.9.0 
    webpack-dev-server: ^3.8.0 => 3.11.2 
    webpack-manifest-plugin: ^4.0.2 => 4.0.2 
    webpack-merge: ^4.2.1 => 4.2.2 
```
    
    
   Webpack config for loading CSS
   
```JavaScript
           {
          test: /\.css$/,
          include: [
            path.resolve(__dirname, 'src/'),
            path.resolve(__dirname, '.storybook/'),
            path.resolve(__dirname, 'node_modules/@orgname/'),
          ],
          exclude: [
            path.resolve(
              __dirname,
              'node_modules/@orgname/subscription-widget',
            ),
          ],
          use: [
            {
              loader: 'style-loader',
            },
            {
              loader: 'css-loader',
              options: {
                url: true,
                importLoaders: 1,
                modules: {
                  localIdentName: '[name]__[local]__[hash:base64:5]',
                },
              },
            },
            {
              loader: 'postcss-loader',
              options: {
                postcssOptions: {
                  ...getPostCssConfig(env.BROKER),
                  config: false,
                },
              },
            },
          ],
        },
        {
          test: /\.css$/,
          exclude: [
            path.resolve(__dirname, 'src/'),
            path.resolve(__dirname, '.storybook/'),
            path.resolve(__dirname, 'node_modules/@orgname/'),
          ],
          include: [
            path.resolve(__dirname, 'node_modules'),
            path.resolve(
              __dirname,
              'node_modules/@orgname/subscription-widget',
            ),
          ],
          use: [{ loader: 'style-loader' }, { loader: 'css-loader' }],
        },
```

**Reply by shirshendubhowmick** [reply]
The package which I was taking about is getting processed via the very first CSS rule mentioned above

**Reply by alexander-akait** [reply]
If you remove added package, all works fine?

**Reply by shirshendubhowmick** [reply]
Yeah even if i downgrade to a lower version (which doesn't include the recent change) it works fine

**Reply by shirshendubhowmick** [reply]
Also one important point here. The same package is getting used in another repo where we are still using webpack 4 and similar CSS loading setup. Everything is working fine there.

It looks like something is going with webpack 5 and the loaders and for some reason its also not producing any logs to debug further.

**Reply by alexander-akait** [reply]
> Yeah even if i downgrade to a lower version (which doesn't include the recent change) it works fine

What do you mean `downgrade`? There are a lot of places for freeze, from `postcss-loader` to `webpack`, do you bundle production or development build?

**Reply by shirshendubhowmick** [reply]
By downgrade I meant if I use the previous version of the package (which doesn't have the change)

I have tried with both production & development mode, the issue is occurring for both of them

**Reply by alexander-akait** [reply]
What is the package?

**Reply by shirshendubhowmick** [reply]
Its an internal package as mentioned earlier, its a CSS library, the package contains only CSS files and some font files and the recent changes were made to some of the CSS files.

**Reply by alexander-akait** [reply]
Can you reproduce the problem with pure/simple webpack configuration and this package?

**Reply by shirshendubhowmick** [reply]
Yes I have created a small repo removed all unnecessary configs. Just using the config as shared above and I am able to reproduce the issue.

This small repo is basically a React Hello World project and just imports only this package at `index.js`

I am seeing exact same issue here.

This time it is stuck and on console I can see only `46% building 2/3 entries 2279/2284 dependencies 600/613 modules`

**Reply by alexander-akait** [reply]
Can you show me small repo, you can invite me in private repo and I will look at the problem

**Reply by shirshendubhowmick** [reply]
Unfortunately, these repos belongs to an org and I don't have access to invite outside collaborator.

```
0.4666666666666667 building 2/3 entries 2285/2285 dependencies 609/614 modules 5 active /Users/shirshendubhowmick/smallcase/webpack/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[1]!/Users/shirshendubhowmick/smallcase/webpack/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[1].use[2]!/Users/shirshendubhowmick/smallcase/webpack/node_modules/@orgname/xpackage/index.css
```

I used the webpack progress plugin to get some more output. `@orgname/xpackage`  is the package I was taking about.

And the log above is the last log I can see on the console. It looks like its stuck at postcss loader. Is there anyway I can get more information ?

Since I am not able to share the repo with you. However I can do some screen sharing with you, if thats something suits for you.

**Reply by alexander-akait** [reply]
What is size of `@orgname/xpackage/index.css`? Can you shared `@orgname/xpackage/index.css` content? For better debug you can use `parallelism: 1`:
```
Tip
Combine profile: true with parallelism: 1 to get correct timings. Note that this will slow down the build as well.

```

https://webpack.js.org/configuration/other-options/#parallelism

**Reply by shirshendubhowmick** [reply]
Thanks with `parallelism: 1` in both repos (actual & test) the build process is stuck at the exact same file, `index.css`

The content of the `index.css` is bunch of imports

```CSS
/*****************************************************/
/* RESET CSS */
/*****************************************************/

@import './reset.css';

/*****************************************************/
/* Fonts */
/*****************************************************/

@import './fonts/fonts.css';

/*****************************************************/
/* Typography */
/*****************************************************/

@import './atoms/typography.css';

/*****************************************************/
/* variables */
/*****************************************************/

@import './atoms/variables.css';

/*****************************************************/
/* Icons */
/*****************************************************/

@import './icons/icons.css';

/*****************************************************/
/* Images */
/*****************************************************/

@import './components/images.css';

/*****************************************************/
/* Grid */
/*****************************************************/

@import './components/grid.css';

/*****************************************************/
/* Spacers */
/*****************************************************/

@import './components/spacers.css';

/*****************************************************/
/* Container */
/*****************************************************/

@import './components/container.css';

/*****************************************************/
/* Utilities */
/*****************************************************/

@import './atoms/utilities.css';
@import './atoms/mobile-utils.css';


/*****************************************************/
/* Bottom fixed bars */
/*****************************************************/

@import './components/bottom-fixed.css';

/*****************************************************/
/* Buttons */
/*****************************************************/

@import './components/buttons.css';

/*****************************************************/
/* Tag */
/*****************************************************/

@import './components/tag.css';

/*****************************************************/
/* Inputs */
/*****************************************************/

@import './components/inputs.css';

/*****************************************************/
/* Cards */
/*****************************************************/

@import './components/card.css';

/*****************************************************/
/* Dropdown */
/*****************************************************/

@import './components/dropdown.css';

/*****************************************************/
/* Tabs */
/*****************************************************/

@import './components/tabs.css';

/*****************************************************/
/* Forms */
/*****************************************************/

@import './components/forms.css';

/*****************************************************/
/* Stat Box */
/*****************************************************/

@import './components/statbox.css';

/*****************************************************/
/* Global Animatiosns */
/*****************************************************/

@import './components/animations.css';

/*****************************************************/
/* Meta Pages */
/*****************************************************/

@import './components/meta.css';

/*****************************************************/
/* Header */
/*****************************************************/

@import './components/header.css';

```

No changes were made to this particular file.

This is how our post css config looks like
```JavaScript
{
    plugins: [
      require('postcss-import'),
      require('postcss-custom-media')({
        importFrom: './node_modules/@orgname/xpackage/resolutions.css',
      }),
      require('autoprefixer')({
        flexbox: 'no-2009',
      }),
      require('postcss-mixins'),
      require('postcss-nested'),
      require('postcss-simple-vars')({
        variables: colors,
      }),
      require('postcss-flexbugs-fixes'),
      require('postcss-custom-properties')({
        importFrom: [
          './node_modules/@orgname/xpackage/variables.css',
        ],
        preserve: false,
      }),
    ],
  }
  ```

**Reply by shirshendubhowmick** [reply]
I fixed issue.

Was postcss-custom-properties v11 which says it has support for postcss v8.

However I saw that there is one more release of v12 which says added support for postcss v8+

Not exactly sure what is meant by that
https://github.com/postcss/postcss-custom-properties/releases

After I upgraded to v12 everything is now working fine.

Still I am not sure what was wrong in v11 and what exactly has changed in v12 that fixed the issue.

**Reply by shirshendubhowmick** [reply]
Thank you for your help and patience @alexander-akait

**Reply by alexander-akait** [reply]
It's my pleasure :smile:

**Comment by shirshendubhowmick** [comment, marked as answer]
This was resolved after updating `postcss-custom-properties` from v11 to v12
